core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/plugins/ai-image-prompt.plugin.ts

```diff
@@ -0,0 +1,175 @@
+import { Node } from 'prosemirror-model';
+import { EditorState, Plugin, PluginKey, Transaction } from 'prosemirror-state';
+import { EditorView } from 'prosemirror-view';
+import { Subject } from 'rxjs';
+import tippy, { Instance, Props } from 'tippy.js';
+
+import { ComponentRef } from '@angular/core';
+
+import { takeUntil } from 'rxjs/operators';
+
+import { Editor } from '@tiptap/core';
+
+import { AIImagePromptComponent } from '../ai-image-prompt.component';
+import { AI_IMAGE_PROMPT_PLUGIN_KEY } from '../ai-image-prompt.extension';
+import { TIPPY_OPTIONS } from '../utils';
+
+interface AIImagePromptProps {
+    pluginKey: PluginKey;
+    editor: Editor;
+    element: HTMLElement;
+    tippyOptions: Partial<Props>;
+    component: ComponentRef<AIImagePromptComponent>;
+}
+
+interface PluginState {
+    open: boolean;
+    form: [];
+}
+
+export type AIImagePromptViewProps = AIImagePromptProps & {
+    view: EditorView;
+};
+
+export class AIImagePromptView {
+    public editor: Editor;
+
+    public node: Node;
+
+    public element: HTMLElement;
+
+    public view: EditorView;
+
+    public tippy: Instance | undefined;
+
+    public tippyOptions: Partial<Props>;
+
+    public pluginKey: PluginKey;
+
+    public component: ComponentRef<AIImagePromptComponent>;
+
+    private destroy$ = new Subject<boolean>();
+
+    constructor(props: AIImagePromptViewProps) {
+        const { editor, element, view, tippyOptions = {}, pluginKey, component } = props;
+
+        this.editor = editor;
+        this.element = element;
+        this.view = view;
+
+        this.tippyOptions = tippyOptions;
+
+        this.element.remove();
+        this.pluginKey = pluginKey;
+        this.component = component;
+
+        this.view.dom.addEventListener('keydown', this.handleKeyDown.bind(this));
+
+        this.component.instance.formSubmission.pipe(takeUntil(this.destroy$)).subscribe(() => {
+            this.editor.commands.closeImagePrompt();
+            this.editor.commands.insertLoaderNode();
+        });
+
+        this.component.instance.aiResponse
+            .pipe(takeUntil(this.destroy$))
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            .subscribe((contentlet: any) => {
```

Comment:
Try to type all `variables`/`HTTP Response` to avoid `TypeScript` and `lint` errors